docs/api/Query.md

```diff
@@ -4,15 +4,17 @@ Dynamoose provides the ability to query a model by using the `Model.query` funct
 
 ## Model.query([filter])
 
-This is the basic entry point to construct a query request. The filter property is optional and can either be an object or a string representing the key you which to first filter on. In the event you don't pass in any parameters and don't call any other methods on the query object it will query with no filters or options.
+This is the basic entry point to construct a query request. It requires to set at least the hashKey (partition key) of the item(s). The filter property is optional and can either be an object or a string representing the key you which to first filter on. In the event you don't pass in any parameters and don't call any other methods on the query object it will query with no filters or options.
 
 ```js
-Cat.query("breed").contains("Terrier").exec() // will query all items and filter all items where the key `breed` contains `Terrier`
-Cat.query({"breed": {"contains": "Terrier"}}).exec() // will query all items and filter all items where the key `breed` contains `Terrier`
+Cat.query("breed").contains("Terrier").exec() // will query all items where the hashKey (partition key) `breed` contains `Terrier`
+Cat.query({"breed": {"contains": "Terrier"}}).exec() // will query all items where the hashKey (partition key) `breed` contains `Terrier`
 ```
 
 If you pass an object into `Model.query` the object for each key should contain the comparison type. For example, in the last example above, `contains` was our comparison type. This comparison type must match one of the comparison type functions listed on this page.
 
+**Please note:** `Model.query()` combines both the `KeyConditionExpression` and the `FilterExpression` from DynamoDB. If you query for an attribute that you defined as your hashKey (partition key) or rangeKey (sort key) DynamoDB will use `KeyConditionExpression`. This is the most performant and cost efficient way and we highly recommend setting hashKey (partition key) and rangeKey (sortKey) that you can query for. If you query for attributes that are not defined as your hashKey (partition key) or rangeKey (sortKey) DynamoDB might select more items at first which could have a bad impact on performance and costs. Also consider using secondary indexes instead.    
```

Comment:
> **Please note:**

Not sure I'm a fan of the bold here. We can add it later if needed. But it adds unnecessary styling, and draws peoples attention to that. If you are new to DynamoDB and don't understand this concept, you should likely be reading most/all of the documentation. I don't think it's necessary to draw additional attention to this point and risk people skipping or skimming over other valuable information.

> This is the most performant and cost efficient way and we highly recommend setting hashKey (partition key) and rangeKey (sortKey) that you can query for.

I don't think this is always true. I can for sure think of legit instances where you don't need to query for a range/sort key. Of course you always need the hash/partition key. But I can totally think of legit instances where just querying the hash/partition key will meet a use case and is the most performant & cost effective solution.

> Also consider using secondary indexes instead.

Something doesn't feel right about this sentence. It feels overly broad. I'm not quite sure how to fix it tho. I don't think it really adds anything to this documentation, and I feel like raises more questions than it answers.

Not quite sure what to suggest changing it to tho. If you have any thoughts I'd love to hear them.

Reply:
I changed to:

> Please note: `Model.query()` combines both the `KeyConditionExpression` and the `FilterExpression` from DynamoDB. If you query for an attribute that you defined as your hashKey (partition key) or rangeKey (sort key) DynamoDB will use `KeyConditionExpression`. This could be the most performant and cost efficient way to query for. If querying for attributes that are not defined as your hashKey (partition key) or rangeKey (sortKey) DynamoDB might select more items at first (that can be found by hashKey and rangeKey) and then filter the result. This could have a bad impact on performance and costs.

What do you think?